src/project-manager.ts

```diff
@@ -796,7 +798,11 @@ export class ProjectConfiguration {
 	 * the whole project or in some files
 	 */
 	syncProgram(): void {
-		this.program = this.getService().getProgram();
+		try {
+			this.program = this.getService().getProgram();
+		} catch (e) {
+			this.logger.error(`Cannot create program object for ${this.rootFilePath}`, e);
```

Comment:
Not die, but may fail the operation. I was worried that if the error is swallowed, the calling function will just continue and return incorrect results or error otherwise, hiding the true error reason.